libraries/azure_network_security_group.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzureNetworkSecurityGroup < AzurermResource
+  name 'azure_network_security_group'
+  desc 'Verifies settings for Network Security Groups'
+  example "
+    describe azure_network_security_group(resource_group: 'example', name: 'name') do
+      its(name) { should eq 'name'}
+    end
+  "
+
+  ATTRS = %i(
+    name
+    id
+    etag
+    type
+    location
+    tags
+    properties
+  ).freeze
+
+  attr_reader(*ATTRS)
+
+  def initialize(resource_group: nil, name: nil)
+    resp = client.network_security_group(resource_group, name)
+    return if resp.nil? || resp.key?('error')
+
+    ATTRS.each do |field|
+      instance_variable_set("@#{field}", resp[field.to_s])
+    end
+
+    @exists = true
+  end
+
+  def to_s
+    "'#{name}' Network Security Group"
+  end
+
+  def security_rules
+    @security_rules ||= @properties['securityRules']
+  end
+
+  def default_security_rules
+    @default_security_rules ||= @properties['defaultSecurityRules']
+  end
+
+  SSH_CRITERIA = %i(ssh_port access_allow direction_inbound tcp source_open).freeze
+  def allow_ssh_from_internet?
+    @allow_ssh_from_internet ||= matches_criteria?(SSH_CRITERIA, security_rules_properties)
+  end
+  RSpec::Matchers.alias_matcher :allow_ssh_from_internet, :be_allow_ssh_from_internet
+
+  RDP_CRITERIA = %i(rdp_port access_allow direction_inbound tcp source_open).freeze
+  def allow_rdp_from_internet?
+    @allow_rdp_from_internet ||= matches_criteria?(RDP_CRITERIA, security_rules_properties)
+  end
+  RSpec::Matchers.alias_matcher :allow_rdp_from_internet, :be_allow_rdp_from_internet
+
+  private
+
+  def security_rules_properties
+    security_rules.collect { |rule| rule['properties'] }
+  end
+
+  def matches_criteria?(criteria, properties)
+    properties.any? { |property| criteria.all? { |method| send(:"#{method}?", property) } }
+  end
+
+  def ssh_port?(properties)
+    matches_port?(destination_port_ranges(properties), '22')
+  end
+
+  def rdp_port?(properties)
+    matches_port?(destination_port_ranges(properties), '3389')
+  end
+
+  def destination_port_ranges(properties)
+    return Array(properties['destinationPortRange']) if properties['destinationPortRanges'].empty?
+    return properties['destinationPortRanges'] if properties['destinationPortRange'].empty?
+    properties['destinationPortRanges'].push(properties['destinationPortRange'])
+  end
+
+  def matches_port?(ports, match_port)
+    return true if ports.detect { |p| p =~ /^(#{match_port}|\*)$/ }
+    ports
+      .select { |port| port.include?('-') }
```

Comment:
Usually I see the second method on the new line. Not sure if it matters though.

Reply:
So something like?
```
ports.select { .... }
         .collect { .... }
         .any? { ... }
```

Reply:
Yip :smile: